base/src/lib.rs

```diff
@@ -512,7 +514,7 @@ pub enum TpmtHa {
 }
 impl TpmtHa {
```

Comment:
How do you feel about replacing calls to `TpmtHa::union_size()` with  `size_of::<TpmuHaUnion>()` directly, and getting rid of this impl block (here and for others)? I think it might be slightly simpler.